dual_net.py

```diff
@@ -90,28 +94,35 @@ def train(self, tf_records, init_from=None, logdir=None, num_steps=None):
                                       train_mode=True, **self.hparams)
             train_tensors = train_ops(
                 input_tensors, output_tensors, **self.hparams)
-            weight_tensors = logging_ops()
+            weight_summary_op = logging_ops()
+            weight_tensors = tf.trainable_variables()
             self.initialize_weights(init_from)
             if logdir is not None:
                 training_stats = StatisticsCollector()
                 logger = tf.summary.FileWriter(logdir, self.sess.graph)
             for i in tqdm(range(num_steps)):
+                if should_log(i):
+                    before_weights = self.sess.run(weight_tensors)
                 try:
                     tensor_values = self.sess.run(train_tensors)
                 except tf.errors.OutOfRangeError:
                     break
+
                 if logdir is not None:
                     training_stats.report(
-                        tensor_values['policy_cost'],
-                        tensor_values['value_cost'],
-                        tensor_values['l2_cost'],
-                        tensor_values['combined_cost'])
-                    if i % 100 == 0 and logdir is not None:
-                        accuracy_summaries = training_stats.collect()
-                        weight_summaries = self.sess.run(weight_tensors)
-                        global_step = tensor_values['global_step']
-                        logger.add_summary(accuracy_summaries, global_step)
-                        logger.add_summary(weight_summaries, global_step)
+                        {k: tensor_values[k] for k in (
+                            'policy_cost', 'value_cost', 'l2_cost',
+                            'combined_cost')})
+                if should_log(i):
+                    after_weights = self.sess.run(weight_tensors)
+                    gradient_summaries = compute_gradient_ratio(
```

Comment:
I would call these weight_update summaries. It would be useful to also track gradient summaries, which can be computed with the following line:
  grads = tf.gradients(loss, weight_tensors)